packages/ovm/src/contracts/ExecutionManager.sol

```diff
@@ -1064,4 +1022,65 @@ contract ExecutionManager is FullStateManager {
             return(codeContractBytecode, _length)
         }
     }
+
+    /********
+    * Utils *
+    ********/
+
+    /**
+     * @notice Initialize a new context, setting the timestamp, queue origin, and gasLimit as well as zeroing out the
+     *         msgSender of the previous context.
+     *         NOTE: this zeroing may not technically be needed as the context should always end up as zero at the end of each execution.
+     * @param _timestamp The timestamp which should be used for this context.
+     * @param _queueOrigin The queue which this context's transaction was sent from.
+     * @param _ovmTxOrigin The tx.origin for the currently executing transaction. It will be ZERO_ADDRESS if it's not an EOA call.
+     */
+    function initializeContext(uint _timestamp, uint _queueOrigin, address _ovmTxOrigin, address _l1MsgSender) internal {
+        // First zero out the context for good measure (Note ZERO_ADDRESS is reserved for the genesis contract & initial msgSender)
+        restoreContractContext(ZERO_ADDRESS, ZERO_ADDRESS);
```

Comment:
Should `restoreContractContext(...)` now also reset `l1MsgSender`?

Reply:
Ah so the L1MessageSender should only be set one time per tx, and we are already setting it in `initializeContext(...)` so that should be sufficient (it resembles `_timestamp` or `_ovmTxOrigin` in that way).

That said, maybe `restoreContractContext(...)` could be renamed to make more explict that we're not restoring the entire context, but just the active contract & message sender